contracts/HATVaultsTokenWrapper.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/utils/SafeERC20.sol";
+
+
+ // Wrap an ERC20 token with other ERC20 token.
+contract HATVaultsTokenWrapper {
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable wrappedToken;
+    address public immutable poolContract;
+     constructor(address _tokenToWrap, address _poolContract) {
```

Comment:
lets add here token symbol,name and decimal .. so it will be manifested in the UI.